src/modules/crudData.js

```diff
@@ -0,0 +1,161 @@
+    updateContent.className = 'update';
+    updateContent.id = newData.index;
+    const inputCheck = document.createElement('input');
+    inputCheck.type = 'checkbox';
+    inputCheck.id = newData.index;
+    inputCheck.className = 'complete';
+    const description = document.createElement('input');
+    description.id = newData.index;
+    description.type = 'text';
+    description.value = newData.description;
+    description.className = 'updateDesc';
+    updateContent.appendChild(inputCheck);
+    updateContent.appendChild(description);
+
+    if (newData.completed === true) {
+      inputCheck.checked = true;
+      description.classList.add('checked');
+    } else {
+      description.classList.remove('checked');
+      inputCheck.checked = false;
+    }
+
+    const iconContent = document.createElement('div');
+    iconContent.className = 'icons';
+    const moveIcon = document.createElement('i');
+    moveIcon.id = 'move';
+    moveIcon.className = 'las la-ellipsis-v';
+    const deleteIcon = document.createElement('i');
+    deleteIcon.id = 'delete';
+    deleteIcon.className = 'las la-trash-alt hidden';
+    iconContent.appendChild(moveIcon);
+    iconContent.appendChild(deleteIcon);
+
+    todoTask.appendChild(updateContent);
+    todoTask.appendChild(iconContent);
+    todoList.appendChild(todoTask);
+
+    // Update the description to the task
+    const edit = todoTask.querySelector('.updateDesc');
+
+    edit.addEventListener('input', () => {
+      newData.description = edit.value;
+      Functionality.updateStorage(newData.index, newData);
+    });
+
+    edit.addEventListener('focus', () => {
+      const allItems = document.querySelectorAll('.task-list, .task');
+      const move = todoTask.querySelector('#move');
+      const trash = todoTask.querySelector('#delete');
+      allItems.forEach((item) => {
+        if (item === todoTask) {
+          item.classList.add('active');
+          trash.classList.remove('hidden');
+          move.classList.add('hidden');
+        } else {
+          item.classList.remove('active');
+          trash.classList.remove('hidden');
+          move.classList.add('hidden');
+        }
+      });
+    });
+
+    // checked if complete is true
+    const checked = todoTask.querySelector('.complete');
+    checked.addEventListener('change', () => {
+      newData.completed = !newData.completed;
+      if (newData.completed === true) {
+        inputCheck.checked = true;
+        description.classList.add('checked');
+        Functionality.deleteComplete();
+      } else {
+        description.classList.remove('checked');
+        inputCheck.checked = false;
+      }
+      Functionality.updateStorage(newData.index, newData);
+    });
+
+    // delete each task of the list
+    const trash = todoTask.querySelector('#delete');
+    trash.addEventListener('click', () => {
+      Functionality.deleteData(newData.index);
+      todoList.removeChild(todoTask);
+    });
```

Comment:
- [ ] Awesome implemeting the changes pointed out earlier, but you are almost there. 👍 I observed that if I deleted an Item from the UI, wrong item would be deleted in the localstorage. Kindly check the screenshot below for better understanding. 👍 

<img width="921" alt="Untitled" src="https://github.com/Fombi-Favour/To-Do-list/assets/103386183/726b7d73-aba5-4072-b108-b7609e166e06">

Reply:
Please I checked it based on what you're saying and it doesn't do that